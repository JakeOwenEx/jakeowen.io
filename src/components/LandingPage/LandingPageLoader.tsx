import * as React from "react";
import { useState } from "react";
import handleViewport from "react-in-viewport";
import { AnimationTimeline } from "../../utils/Animations";

export const LandingPageLoader = (): DocumentTimeline => {
  const animationTl = new AnimationTimeline("linear");
  return animationTl
    .addFadeIn("name-text", 50, 700)
    .addFadeIn("menu-selection", 50, 1000)
    .addFadeIn("get-in-touch", -50, 500)
    .addFadeIn("GitHubLogo", 50, 300)
    .addFadeIn("MediumLogo", 50, 300)
    .addFadeIn("LinkedInLogo", 50, 300)
    .addFadeIn("social-icon-grey-bar", 50, 300)
    .getTimeline();
};

const SectionLoader = (cssClass: string): DocumentTimeline => {
  const animationTl = new AnimationTimeline("linear");
  return animationTl
    .addFadeIn(`${cssClass}-text-title`, 50, 700)
    .addFadeIn(`${cssClass}-box`, 50, 1000)
    .getTimeline();
};

interface Loader {
  cssClass: string;
  setHasBeenVisible(opacity: number): void;
  inViewport?: boolean;
  forwardedRef?: string;
}

const SectionLoaderViewport: React.FC<Loader> = (props) => {
  const { inViewport, forwardedRef } = props;
  const [animated, setAnimated] = useState(false);
  if (inViewport && !animated) {
    SectionLoader(props.cssClass);
    setAnimated(true);
    props.setHasBeenVisible(1);
  }
  return <div ref={forwardedRef} />;
};
export const SectionLoaderComponent = handleViewport(SectionLoaderViewport);
