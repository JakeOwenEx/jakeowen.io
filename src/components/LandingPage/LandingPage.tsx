import * as React from "react";
import MenuSelection from "./MainMenu/Menu";
import { LandingPageLoader } from "./LandingPageLoader";
import { scaleAnimation } from "../../utils/Animations";
import { useState } from "react";

const NameTitle: React.FC = () => {
  return (
    <div className="name-text">
      <h1 className="name-title">Jake Owen.</h1>
      <h1 className="strap-line">Software Engineer.</h1>
    </div>
  );
};

const GetInTouchBox: React.FC = () => {
  const cssClass = "get-in-touch";
  const [backgroundColour, setBackgroundColour] = useState("#2B3E4D");
  function scaleAndUpdateBackgroundColour(
    scale: number,
    backgroundColour: string,
  ) {
    scaleAnimation(scale, cssClass);
    setBackgroundColour(backgroundColour);
  }
  return (
    <button
      onMouseOver={() => scaleAndUpdateBackgroundColour(1.05, "#32424F")}
      onMouseOut={() => scaleAndUpdateBackgroundColour(1, "#2B3E4D")}
      className={cssClass}
      style={{ backgroundColor: backgroundColour }}
    >
      Get In Touch
    </button>
  );
};

const LandingPage = () => {
  const [pageLoaded, setPageLoaded] = useState(false);
  React.useEffect(() => {
    if (!pageLoaded) {
      LandingPageLoader();
      setPageLoaded(true);
    }
  });

  return (
    <div className="landingPage">
      <GetInTouchBox />
      <div className="container name-container name">
        <NameTitle />
        <MenuSelection />
      </div>
    </div>
  );
};

export default LandingPage;
