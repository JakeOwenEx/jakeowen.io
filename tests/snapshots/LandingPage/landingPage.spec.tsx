import * as React from "react";
import { create } from "react-test-renderer";
import LandingPage from "../../../src/components/LandingPage/LandingPage";

describe("Snapshot Sections", () => {
  it("Landing Page", () => {
    expect(create(<LandingPage />)).toMatchSnapshot();
  });
});
