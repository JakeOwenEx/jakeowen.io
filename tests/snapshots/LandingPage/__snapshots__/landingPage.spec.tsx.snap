// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Sections Landing Page 1`] = `
<div
  className="landingPage"
>
  <button
    className="get-in-touch"
    onMouseOut={[Function]}
    onMouseOver={[Function]}
    style={
      Object {
        "backgroundColor": "#2B3E4D",
      }
    }
  >
    Get In Touch
  </button>
  <div
    className="container name-container name"
  >
    <div
      className="name-text"
    >
      <h1
        className="name-title"
      >
        Jake Owen.
      </h1>
      <h1
        className="strap-line"
      >
        Software Engineer.
      </h1>
    </div>
    <div
      className="row menu-container"
      style={
        Object {
          "marginTop": "5%",
        }
      }
    >
      <div
        className="col-md-5"
      >
        <button
          className="menu-selection menu-selection-about-me"
          onClick={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
        >
          <span
            className="neon-blue"
          >
            1
            . 
          </span>
          <span
            style={
              Object {
                "color": "#ABB4BB",
              }
            }
          >
            About Me
          </span>
          <span
            className="neon-blue"
          >
            .
          </span>
        </button>
        <button
          className="menu-selection menu-selection-projects"
          onClick={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
        >
          <span
            className="neon-blue"
          >
            3
            . 
          </span>
          <span
            style={
              Object {
                "color": "#ABB4BB",
              }
            }
          >
            Projects
          </span>
          <span
            className="neon-blue"
          >
            .
          </span>
        </button>
      </div>
      <div
        className="col-md-5"
      >
        <button
          className="menu-selection menu-selection-resume"
          onClick={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
        >
          <span
            className="neon-blue"
          >
            2
            . 
          </span>
          <span
            style={
              Object {
                "color": "#ABB4BB",
              }
            }
          >
            Resume
          </span>
          <span
            className="neon-blue"
          >
            .
          </span>
        </button>
        <button
          className="menu-selection menu-selection-articles"
          onClick={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
        >
          <span
            className="neon-blue"
          >
            4
            . 
          </span>
          <span
            style={
              Object {
                "color": "#ABB4BB",
              }
            }
          >
            Articles
          </span>
          <span
            className="neon-blue"
          >
            .
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;
